### ISSUE TITLE
`no-misused-generics`: false negative

### ISSUE THREAD

**Issue by OliverJAsh** [cartant/eslint-plugin-etc#30]
```ts
// Errors as expected
type MyFunction = <T>(param: T) => number;

// Expected error, got none
const myFunction = <T>(param: T) => 1;
```

**Comment by cartant** [comment]
If you have TypeScript configured to parse JSX, the RHS of that variable declaration will be parsed as a `JsxElement` and will be ignored by the rule. Try:

```tsx
const myFunction = <T,>(param: T) => 1;
//                   ^
```

**Comment by OliverJAsh** [comment]
Unfortunately that doesn't seem to help. Reduced test case: https://github.com/OliverJAsh/eslint-no-misused-generics-test

**Comment by cartant** [comment]
Not going to fix this 'cause I don't have the time to figure out why it's doing what it's doing:

https://github.com/cartant/eslint-plugin-etc/blob/4152f18e29e8c3b945786f58cc76a406edb5baba/tests/rules/no-misused-generics.ts#L105-L117

I didn't write this rule; I just ported it. And, TBH, I'm beginning to think that it's more trouble than it's worth. It's too simplistic and won't work with valid, but context-sensitive generic usage. I'm inclined to delete it and give the package a major version bump.

**Comment by cartant** [comment]
Thinking about it some more, my guess is that the original author made the (correct) decision that it's too difficult to determine whether what's returned relates to the parameter with the type parameter. E.g. this would be okay:

```ts
const myFunction = <T>(param: T) => param;
```

**Comment by OliverJAsh** [comment]
You're totally right. I actually filed the same issue back when we were using Wotan 🤦. Sorry! https://github.com/fimbullinter/wotan/issues/690
